guides/source/form_helpers.md

```diff
@@ -871,8 +871,8 @@ Which will result in a `params` hash that looks like:
 
 All of the form inputs map to the `"person"` hash because we called `fields_for`
 on the `person_form` form builder. By specifying an `:index` option, we mapped
-the address inputs to `person[address][#{address.id}][city]` instead of
-`person[address][city]`. Thus we are able to determine which Address records
+the address inputs to `"person[address][#{address.id}][city]"` instead of
+`person[address][city]`. The value address id here is dynamic. Thus we are able to determine which Address records
```

Comment:
> In my opinion string interpolation could only be done within double quotes, So we should add the whole thing in the double quote.

As I mentioned, there are other places where we use this notation.  For example, [here](https://github.com/rails/rails/blob/51e9fa9bf9701672bc45ab6da4f39212af39db59/guides/source/form_helpers.md?plain=1#L703) and [here](https://github.com/rails/rails/blob/51e9fa9bf9701672bc45ab6da4f39212af39db59/guides/source/7_0_release_notes.md?plain=1#L155-L158).  It is a convenient way to represent value substitution in general, and it should be familiar to Ruby programmers.

By adding double quotes around `person[address][#{address.id}][city]`, it no longer matches the format of `person[address][city]` (or any other such attribute values in the guide).

Also, I feel like the sentence "The value address id here is dynamic." is redundant with the very notation we are using.

If this paragraph is not clear, perhaps we should reword it, instead of fiddling with notation.

Would the following be more understandable?

```markdown
All of the form inputs map to the `"person"` hash because we called `fields_for`
on the `person_form` form builder. Also, by specifying `index: address.id`, we
rendered the `name` attribute of each city input as `person[address][#{address.id}][city]`
instead of `person[address][city]`. Thus we are able to determine which Address
records should be modified when processing the `params` hash.
```

(Also, whenever the changes are finalized, please [squash the commits](https://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits).)

Reply:
@jonathanhefner make sense to me. 
I have changed it. Thanks.